starters/express-typescript-typeorm-postgres/src/interfaces/results.ts

```diff
@@ -0,0 +1,11 @@
+import { ERROR_TYPE, SUCCESS_TYPE } from './constants';
+
+export interface SuccessResult<T> {
+  type: typeof SUCCESS_TYPE;
+  data: T;
+}
+
+export interface ErrorResult {
+  type: typeof ERROR_TYPE;
+  message: string;
+}
```

Comment:
This is not what I intended with this, sorry but I won't change it.

Reply:
I think have the `Result` types be 2 distinct types, the `SuccessResult` and the `ErrorResult`, is better and they can be discriminated by the `type` property if we need a [discriminated union type](https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html#discriminating-unions)
```ts
type Result = SuccessResult | ErrorResult
```

But that being said, do we really need the constants here?  We can just have the string literal values:

```ts
export interface SuccessResult<T> {
  type: 'success';
  data: T;
}

export interface ErrorResult {
  type: 'error';
  message: string;
}
```

...that is if we really need these result types at all

Reply:
The constants are for avoiding typos in the future. I hate string literals throughout a codebase, if the developers need to type out `result.type === 'success'` every time, it is inevitable in the future to have bugs coming from `result.type === 'scucess'`. My thinking here with the constants that we should have these strings declared at one place.

I do believe that having this type of setup is a best practice, because it communicates intent and makes the codebase manageable.